cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -154,21 +157,44 @@ func NewClusterStateRegistry(cloudProvider cloudprovider.CloudProvider, config C
 	}
 }
 
-// RegisterScaleUp registers scale up.
-func (csr *ClusterStateRegistry) RegisterScaleUp(request *ScaleUpRequest) {
+// RegisterOrUpdateScaleUp registers scale-up for give node group or changes requested node increase
+// count.
+// If delta is positive then number of new nodes requested is increased; Time and expectedAddTime
+// are reset.
+// If delta is negative the number of new nodes requested is decreased; Time and expectedAddTime are
+// left intact.
+func (csr *ClusterStateRegistry) RegisterOrUpdateScaleUp(nodeGroup cloudprovider.NodeGroup, delta int, currentTime time.Time) {
 	csr.Lock()
 	defer csr.Unlock()
+	csr.registerOrUpdateScaleUpNoLock(nodeGroup, delta, currentTime)
+}
 
-	oldScaleUpRequest, found := csr.scaleUpRequests[request.NodeGroup.Id()]
+func (csr *ClusterStateRegistry) registerOrUpdateScaleUpNoLock(nodeGroup cloudprovider.NodeGroup, delta int, currentTime time.Time) {
+	scaleUpRequest, found := csr.scaleUpRequests[nodeGroup.Id()]
 	if !found {
-		csr.scaleUpRequests[request.NodeGroup.Id()] = request
+		scaleUpRequest = &ScaleUpRequest{
+			NodeGroup:       nodeGroup,
+			Increase:        delta,
```

Comment:
You can actually register a scale-up with negative delta here. I'm fairly certain this would lead to something bad happening down the line (messed up acceptableRanges calculation? negative scale-up timing out triggering backoff?).